lib/iex/endpoints/historial_prices.rb

```diff
@@ -0,0 +1,29 @@
+module IEX
+  module Endpoints
+    module HistoricalPrices
+      def historical_prices(symbol, options = {})
+        if options[:range] == 'date'
+          raise ArgumentError unless options[:date].present?
+          raise ArgumentError, 'Date param must be a Date object' unless options[:date].class == Date
```

Comment:
`.is_a?(Date)` is the proper way to do this, for one it can inherit from `Date`

I think you want to support non-Date input and pass it through in that case. So I would just drop this case. When passing `Date`, call `strftime`, otherwise pass through the value as is.

```ruby
options = options.dup
options[:date] = options[:date].strftime('...') if options[:date].is_a?(Date)
```

Reply:
Good point about `is_a` thanks.

Fair enough with passing the date through as a string. If the string date is incorrect then the call with simply error out. 👍